frontend-v2/src/app/users/user-form.tsx

```diff
@@ -0,0 +1,347 @@
+            form.handleSubmit()
+          }}
+          className="space-y-6"
+        >
+          <div className="flex items-center justify-end">
+            <Button type="submit" disabled={isSubmitting}>
+              {isSubmitting ? (
+                <>
+                  <Loader2 className="h-4 w-4 animate-spin mr-2" />
+                  Saving...
+                </>
+              ) : (
+                'Save user'
+              )}
+            </Button>
+          </div>
+
+          <div className="grid gap-4 md:grid-cols-2">
+            <form.Field name="name">
+              {(field) => (
+                <div className="space-y-1">
+                  <Label htmlFor="user-name">Name</Label>
+                  <Input
+                    id="user-name"
+                    value={field.state.value}
+                    onChange={(e) => field.handleChange(e.target.value)}
+                    onBlur={field.handleBlur}
+                    placeholder="Full name"
+                  />
+                  {field.state.meta.errors[0] && (
+                    <p className="text-sm text-destructive">
+                      {field.state.meta.errors[0]}
+                    </p>
+                  )}
+                </div>
+              )}
+            </form.Field>
+
+            <form.Field name="email">
+              {(field) => (
+                <div className="space-y-1">
+                  <Label htmlFor="user-email">Email</Label>
+                  <Input
+                    id="user-email"
+                    type="email"
+                    value={field.state.value}
+                    onChange={(e) => field.handleChange(e.target.value)}
+                    onBlur={field.handleBlur}
+                    placeholder="user@example.org"
+                  />
+                  {field.state.meta.errors[0] && (
+                    <p className="text-sm text-destructive">
+                      {field.state.meta.errors[0]}
+                    </p>
+                  )}
+                </div>
+              )}
+            </form.Field>
+
+            <form.Field name="chapterId">
+              {(field) => (
+                <div className="space-y-1">
+                  <Label htmlFor="user-chapter">Chapter</Label>
+                  <Select
+                    value={
+                      field.state.value !== null
+                        ? String(field.state.value)
+                        : undefined
+                    }
+                    onValueChange={(value: string) => {
+                      const next = parseInt(value, 10)
+                      field.handleChange(Number.isNaN(next) ? null : next)
+                    }}
+                    disabled={chaptersLoading}
+                  >
+                    <SelectTrigger id="user-chapter">
+                      <SelectValue placeholder="Select a chapter" />
+                    </SelectTrigger>
+                    <SelectContent>
```

Comment:
the page does a weird scroll jump when this opens. not blocking on it now, but we should try to fix it at some point.